types_test.go

```diff
@@ -18,12 +18,10 @@ func TestNumericDate(t *testing.T) {
 
 	jwt.TimePrecision = time.Microsecond
 
-	raw := `{"iat":1516239022,"exp":1516239022.12345}`
+	raw := `{"iat":1516239022.000000,"exp":1516239022.123450}`
```

Comment:
I suppose this makes sense. 

I personally haven't encountered a need to set the precision in any of the projects I've worked on, and second granularity (without precision) has been sufficient.

Maybe we're overthinking it, and the alternative solution is to _always_ set the default precision to 0.

Reply:
Yes it does make sense. We are setting precision to microsecond and are serialising as microsecond. Sounds good for me.